File: app/src/main/java/com/bigscreen/mangindo/network/model/response/NewReleaseResponse.java
package com.bigscreen.mangindo.network.model.response;

import android.os.Parcel;
import android.os.Parcelable;

import com.bigscreen.mangindo.network.model.Manga;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class NewReleaseResponse implements Parcelable {

    @SerializedName("komik")
    private List<Manga> comics;

    public List<Manga> getComics() {
        return comics;
    }

    public void setMangas(List<Manga> komik) {
        this.comics = komik;
    }

    public NewReleaseResponse() {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.comics);
    }

    protected NewReleaseResponse(Parcel in) {
        this.comics = in.createTypedArrayList(Manga.CREATOR);
    }

    public static final Parcelable.Creator<NewReleaseResponse> CREATOR = new Parcelable.Creator<NewReleaseResponse>() {
        @Override
        public NewReleaseResponse createFromParcel(Parcel source) {
            return new NewReleaseResponse(source);
        }

        @Override
        public NewReleaseResponse[] newArray(int size) {
            return new NewReleaseResponse[size];
        }
    };

    @Override
    public String toString() {
        return "NewReleaseResponse{" +
                "comics=" + comics +
                '}';
    }
}

File: app/src/main/java/com/bigscreen/mangindo/network/model/response/NewReleaseResponse.kt
package com.bigscreen.mangindo.network.model.response

import android.os.Parcel
import android.os.Parcelable
import com.bigscreen.mangindo.network.model.Manga
import com.google.gson.annotations.SerializedName


data class NewReleaseResponse(@SerializedName("komik")
                              var comics: List<Manga> = arrayListOf()
) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.createTypedArrayList(Manga))

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeTypedList(comics)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<NewReleaseResponse> {
        override fun createFromParcel(parcel: Parcel): NewReleaseResponse {
            return NewReleaseResponse(parcel)
        }

        override fun newArray(size: Int): Array<NewReleaseResponse?> {
            return arrayOfNulls(size)
        }
    }
}
